collated/main/A0126240W.md

```diff
@@ -0,0 +1,1468 @@
+            + "Parameters: todo/schedule INDEX (must be a positive integer) description/date/time/tag DESCRIPTION/DATE/TIME/TAG\n"
+            + "Example: " + COMMAND_WORD + " todo 1 tag priority low";
+    
+    public static final String MESSAGE_UPDATE_TASK_SUCCESS = "Updated Task: %1$s";
+    public static final String MESSAGE_DUPLICATE_TASK = "This task already exists in WhatNow";
+    
+    private static final String ARG_TYPE_DESCRIPTION = "description";
+    private static final String ARG_TYPE_DATE = "date";
+    private static final String ARG_TYPE_TIME = "time";
+    private static final String ARG_TYPE_TAG = "tag";
+    private static final String DELIMITER_BLANK_SPACE = " ";
+    private static final String TASK_TYPE_TODO = "todo";
+    private static final String TASK_TYPE_FLOATING = "floating";
+    private static final String TASK_TYPE_NOT_FLOATING = "not_floating";
+    private static final String DEFAULT = "default";
+    
+    private static final int ZERO = 0;
+    private static final int ONE = 1;
+    private static final int TWO = 2;
+    
+    public final int targetIndex;
+    public final String taskType;
+    public final String arg_type;
+    public final String arg;
+    private Task toUpdate;
+    
+    public UpdateCommand(String taskType, int targetIndex, String arg_type, String arg) throws IllegalValueException, ParseException {
+        this.taskType = taskType;
+        this.targetIndex = targetIndex;
+        this.arg_type = arg_type;
+        this.arg = arg;
+        processArg();
+    }
+    
+    /**
+     * Processes the arguments in the update command
+     *
+     * @throws IllegalValueException if any of the raw values are invalid
+     * @throws ParseException 
+     */
+    private void processArg() throws IllegalValueException, ParseException {
+        String newName = DEFAULT;
+        String date = null;
+        String startDate = null;
+        String endDate = null;
+        String time = null;
+        String startTime = null;
+        String endTime = null;
+        final Set<Tag> tagSet = new HashSet<>();
+        
+        if (arg_type.toUpperCase().compareToIgnoreCase(ARG_TYPE_DESCRIPTION) == ZERO) {
+            newName = arg;
+        }
+        
+        if (arg_type.toUpperCase().compareToIgnoreCase(ARG_TYPE_DATE) == ZERO) {
+            if (arg != null) {
+                String[] argComponents = arg.trim().split(DELIMITER_BLANK_SPACE);
+
+                if (argComponents.length == ONE) {
+                    date = argComponents[ZERO];  
+                } else if (argComponents.length == TWO) {
+                    startDate = argComponents[ZERO];
+                    endDate = argComponents[ONE];
+                }
+            } 
+        }
+        
+        if (arg_type.toUpperCase().compareToIgnoreCase(ARG_TYPE_TIME) == ZERO) {
+            if (arg != null) {
```

Comment:
check for this first and get it out of the way; later can do all the important stuff this method does.
you will reduce the nesting level as well.
please refer to the handouts from one of the first few lectures that talks about this